src/test/java/org/zeroturnaround/zip/SkipForJava8Runner.java

```diff
@@ -0,0 +1,43 @@
+package org.zeroturnaround.zip;
+
+import java.util.zip.ZipEntry;
+
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.InitializationError;
+
+/**
+ * JUnit runner that can be used to skip tests for JDK8. This is needed
+ * for example some timestamp preserving tests that should have custom
+ * JDK 8 implementations and custom pre JDK8 tests.
+ */
+public class SkipForJava8Runner extends BlockJUnit4ClassRunner {
```

Comment:
I didn't know that rules can be used like that. Btw, did you check out the source from that article? This is what it proposes https://gist.github.com/rherrmann/7447571 Seems like an overkill compared to the custom runner?